buyer-api/src/facades/addNotariesToOrderFacade.js

```diff
@@ -1,93 +1,136 @@
-import notaryService from '../services/notaryService';
-import { logger, web3, dataExchange, DataOrderContract } from '../utils';
-import { coercion, coin, collection } from '../utils/wibson-lib';
+import {
+  getTransactionReceipt,
+  sendTransaction,
+  retryAfterError,
+} from './helpers';
+import { signingService, notaryService } from '../services';
+import { logger, web3, DataOrderContract } from '../utils';
+import { fromWib } from '../utils/wibson-lib/coin';
 import config from '../../config';
 
-const { isPresent } = coercion;
-const { fromWib } = coin;
-const { partition } = collection;
-
-const buildNotariesParameters = async (
-  notaries,
-  buyerAddress,
-  orderAddress,
-) => {
+const buildNotariesParameters = async (notaries, buyerAddr, orderAddr) => {
   const promises = notaries.map(async ({ notary, publicUrls: { api } }) => {
     const { notarizationFee, ...response } = await notaryService
-      .consent(api, { buyerAddress, orderAddress });
+      .consent(api, { buyerAddress: buyerAddr, orderAddress: orderAddr });
 
     return { ...response, notarizationFee: fromWib(notarizationFee), notary };
   });
 
   return Promise.all(promises);
 };
 
-const takeOnlyNotariesToAdd = async (orderAddress, addresses) => {
+/**
+ * @param {String} orderAddress DataOrder's ethereum address
+ * @param {Array} addresses Array of notary addresses
+ * @returns {Response} The result of the operation.
+ */
+const takeOnlyNotariesToAdd = (orderAddress, addresses) => {
   const dataOrder = DataOrderContract.at(orderAddress);
 
-  let notariesNotAdded = [];
-
-  // eslint-disable-next-line no-restricted-syntax
-  for (const address of addresses) {
-    // eslint-disable-next-line no-await-in-loop
-    const added = await dataOrder.hasNotaryBeenAdded(address);
-
-    if (!added) {
-      notariesNotAdded = [...notariesNotAdded, address];
+  return addresses.reduce((accumulator, address) => {
+    if (!dataOrder.hasNotaryBeenAdded(address)) {
+      return [...accumulator, address];
     }
-  }
-
-  return notariesNotAdded;
+    return accumulator;
+  }, []);
 };
 
-const addNotaryToOrder = async (notaryParameters, buyerAddress) => {
+/**
+ * @async
+ * @param {Object} params Transaction parameters
+ * @param {String} buyerAddress Buyer's ethereum address
+ * @param {Function} addNotaryToOrderSent Callback to notify that tx's been sent
+ * @throws {Error} when AddNotaryToOrder transaction can't be sent
+ * @returns {Response} The result of the operation.
+ */
+const addNotaryToOrder = async (params, buyerAddress, addNotaryToOrderSent) => {
   try {
-    const { logs } = await performTransaction(
+    const dataOrder = DataOrderContract.at(params.orderAddress);
+    if (dataOrder.hasNotaryBeenAdded(params.notary)) {
+      return;
+    }
+
+    const receipt = await sendTransaction(
       web3,
       buyerAddress,
       signingService.signAddNotaryToOrder,
-      notaryParameters,
+      params,
       config.contracts.gasPrice.standard,
```

Comment:
Let's use the fast one here as well. The standard has a huge time variance right now